server/controllers/conversation.js

```diff
@@ -0,0 +1,116 @@
+const User = require("../models/User");
+const Conversation = require("../models/Conversation");
+const Message = require("../models/Message");
+
+// @route POST /start
+// @desc start a new conversation with the logged in user and another valid user
+// @access Private
+exports.startConversation = async (req, res) => {
+  const userId = req.user.id;
+  const { converser } = req.body;
```

Comment:
Optional: You can validate body/path parameters

Ditto throughout!

Reply:
Added a body and param validator middleware using express-validator. Thanks for the suggestion!